Great-Sun-Group/vimbisopay
# Update theme

# Project Cleanup and Configuration Updates

## Overview
This PR focuses on streamlining the project by removing unused features, simplifying configurations, and updating dependencies to create a more maintainable codebase.

## Motivation
The project had accumulated several unused features, complex configurations, and outdated dependencies that were making maintenance more difficult. This cleanup will help:
- Reduce the project's complexity
- Make the codebase more maintainable
- Simplify future development efforts
- Remove potential security risks from unused sensitive configurations

## Changes

### Dependency Management
- Simplified pubspec.yaml configuration
- Removed unused dependencies:
  - smooth_page_indicator
  - shared_preferences
  - url_launcher
  - http
  - local_auth
  - flutter_secure_storage
  - pin_code_fields
  - country_picker
  - sqflite
  - encrypt
  - crypto
- Updated Flutter SDK requirements
- Removed flutter_launcher_icons configuration

### Android Configuration
- Modernized build.gradle to use newer plugin syntax
- Simplified MainActivity implementation by removing unnecessary overrides
- Removed custom splash screen configuration
- Updated app icons to default Flutter icons
- Streamlined AndroidManifest.xml
- Updated Gradle wrapper configuration

### Code Cleanup
- Removed unused authentication-related code:
  - Biometric authentication
  - PIN-based security
  - Security setup screens
- Removed unused features:
  - Dashboard functionality
  - Account management
  - User entity and related implementations
- Removed unused services:
  - EncryptionService
  - NetworkLogger
  - SecurityService
  - DatabaseHelper
- Removed custom theme configurations (AppColors)

### Configuration Files
- Removed sensitive configuration templates:
  - api_config.dart.example
  - encryption_config.dart.example
- Removed VSCode specific settings
- Updated .gitignore to be more concise
- Simplified analysis_options.yaml

### iOS Configuration
- Simplified Info.plist
- Removed Face ID usage description
- Cleaned up unused capabilities

## Testing Considerations
- Basic app functionality should be tested to ensure it still works as expected
- App launch and navigation should be verified
- Build process should be tested for both Android and iOS
- Verify that removing biometric/security features hasn't left any broken flows

## Migration Notes
If you're currently using any of the removed features, you'll need to:
1. Update any screens that were using the removed services
2. Implement alternative authentication if needed
3. Update any configuration files that were extending the removed templates

## Screenshots
![WhatsApp Image 2024-12-16 at 10 15 34 (1)](https://github.com/user-attachments/assets/14e7ef70-c897-4b3c-ad2f-d2940660b379)
![WhatsApp Image 2024-12-16 at 10 15 34 (2)](https://github.com/user-attachments/assets/50dcc36b-6dad-42ca-aaf7-e52696a84885)
![WhatsApp Image 2024-12-16 at 10 15 34 (3)](https://github.com/user-attachments/assets/d43d2020-b078-400f-a4d5-c5ab13fe2329)
![WhatsApp Image 2024-12-16 at 10 15 34 (4)](https://github.com/user-attachments/assets/04c30f7a-a0fd-490a-994d-74c3e7d1ff4d)
![WhatsApp Image 2024-12-16 at 10 15 34 (5)](https://github.com/user-attachments/assets/935d1ee3-d37a-463f-8f0f-10f787786c98)
![WhatsApp Image 2024-12-16 at 10 15 34](https://github.com/user-attachments/assets/2712b50b-eaf7-409f-8ece-50aacafeeb42)
![WhatsApp Image 2024-12-16 at 10 15 35 (1)](https://github.com/user-attachments/assets/9bbf1d44-17f1-408b-ace8-866f3d782e72)

![WhatsApp Image 2024-12-16 at 10 15 35](https://github.com/user-attachments/assets/8cd7c46f-4628-4dbc-8f21-517aec213511)
